CONTRIBUTING.md

```diff
@@ -77,6 +78,64 @@ to do this is to invoke `rustc` like this:
 $ RUST_BACKTRACE=1 rustc ...
 ```
 
+## The Build System
+
+The build system for Rust is complex. It covers bootstrapping the compiler,
+running tests, building documentation and more.
```

Comment:
This should have occurred to me sooner, but since this section spends most of its text on how scary the build system is, perhaps the opening paragraph should not mention the complexity at all and instead include some re-assuring words. Something like

> Rust's build system allows you to bootstrap the compiler, run tests and benchmarks, generate documentation, install your freshly-built rust, and more. It's your best friend for compiling and testing your contributions.